packages/istanbul-lib-instrument/src/visitor.js

```diff
@@ -632,6 +632,13 @@ function programVisitor(types, sourceFilePath = 'unknown.js', opts = {}) {
                     sourceMappingURL: visitState.sourceMappingURL
                 };
             }
+            const { inputSourceMap } = coverageData;
+            // ensure inputSourceMap is a plain object
+            if (inputSourceMap) {
+                coverageData.inputSourceMap = JSON.parse(
```

Comment:
It could be a performance kill if the `inputSourceMap` is big. @coreyfarrell WDYT? Are we going to print `inputSourceMap` to the AST?

Reply:
@JLHwung sorry it's taken me so long to respond.  We do print `inputSourceMap` to AST, this is used to remap coverage data to the original source.  Source mapping is very difficult and we've had (continue to have) many issues with people running mismatched transpile settings so we save the input source map to the specific coverage data.

I'm a bit confused as coverageData.inputSourceMap should always be a plain object.  Do you know how this is not the case?  My preference would be to fix the bug which allows inputSourceMap to not be a plain object.  In theory we could have `inputSourceMap` just be a string to reduce AST memory overhead.  That would be a breaking change though, maybe something we could consider when babel 8 is available (the plan is for istanbuljs next majors upgrade to babel 8).

Reply:
> coverageData.inputSourceMap should always be a plain object

I agree. But since `inputSourceMap` is in our API, I suggest we should validate if it is a plain object before it is consumed later, and finally thrown when instrumenting, which could be quite confusing: At that time we have lost the reason of why `t.valueToNode` is passed with a non-plain object.

`vue-jest` used to pass a `SourceMapGenerator` instance instead of a plain json object (fixed in https://github.com/vuejs/vue-jest/pull/275). This PR was created then. I am good with closing this PR and instead we validate `inputSourceMap` in next major.

Reply:
I'm unclear are you suggesting babel 8 will validate `inputSourceMap` or that `istanbul-lib-instrument` should validate it?

Reply:
I think it would be nice for `istanbul-lib-instrument` to validate it since Babel is not the only downstream users. But yeah Babel 8 could validate it, too.

Reply:
Sorry for the slow response.  I don't object to adding a check if I got a PR for it.  I think we should probably just throw an error in that situation (this should be non-breaking since it will only effect code that is already broken?).  I don't think istanbul-lib-instrument should be doing JSON.parse / JSON.stringify to fix broken input.

One thing I should mention my plan is for the next semver-major of babel-plugin-istanbul not use istanbul-lib-instrument, istanbul-lib-instrument will use babel-plugin-istanbul.  The branch I'm working on is https://github.com/istanbuljs/babel-plugin-istanbul/tree/next.  Still have to experiment with migrating istanbul-lib-instrument to use this updated branch, I'm sure I'll find some minor issues with the current babel plugin when I do that.